src/indentblock.js

```diff
@@ -0,0 +1,289 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module indent-block/indentblock
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+
+import IndentBlockCommand from './indentblockcommand';
+import IndentUsingOffset from './indentcommandbehavior/indentusingoffset';
+import IndentUsingClasses from './indentcommandbehavior/indentusingclasses';
+
+/**
+ * The block indentation feature.
+ *
+ * It registers the `'indentBlock'` and `'outdentBlock'` commands.
+ *
+ * If the plugin {@link module:core/indent/indent~Indent} is defined it also attaches the `'indentBlock'` and `'outdentBlock'` commands to
+ * `'indent'` and `'outdent'` commands.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class IndentBlock extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	constructor( editor ) {
+		super( editor );
+
+		editor.config.define( 'indentBlock', {
+			offset: 1,
+			unit: 'em'
+		} );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	static get pluginName() {
+		return 'IndentBlock';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const editor = this.editor;
+		const schema = editor.model.schema;
+		const conversion = editor.conversion;
+		const configuration = editor.config.get( 'indentBlock' );
+
+		// TODO: supported in CKE4:
```

Comment:
I think it's fine what we have for now. We can document how to enable indent on other elements (which is done by the extending the schema, so it's pretty straightforward).